static/js/keydown_util.ts

```diff
@@ -28,13 +28,13 @@ export function handle(opts: {
     };
 }): void {
     opts.elem.on("keydown", (e) => {
-        const key = e.which || e.keyCode;
+        const key = e.key;
 
         if (e.altKey || e.ctrlKey || e.shiftKey) {
             return;
         }
 
-        function valid_key(key: number): key is keyof Keys {
+        function valid_key(key: string): key is keyof Keys {
```

Comment:
I think this migration is a good idea; we also will want to do it in `hotkey.js` and everywhere else that we do `e.keyCode` in the codebase, so maybe worth forking off a separate series of PRs for that migration in parallel with your typescript work (I'd do one commit per smaller module, and then think hard about doing `hotkey.js`).

The one thing we need to be really careful about in said migration is altKey/ctrlKey/shiftKey handling -- while this module doesn't use it, the semantics of what happens when you type `R` via the `Shift+R` keys are different.  

I think the semantics changes will end up fixing https://github.com/zulip/zulip/issues/18495 if we're thoughtful.